### DISCUSSION TITLE
Data replication in quorum AKS pods

### DISCUSSION THREAD

**Discussion by rushikeshacharya** [ConsenSys/quorum#1486] [Q&A]
#### System information

**Quorum version**: v22.4.0

#### Expected behaviour

Transaction data including events should get replicated to all the permissioned nodes in the network.

#### Actual behaviour
I have setup a 2 org private network (IBFT Consensus) on AKS. Org A has 3 Transaction , 3 Tessera & 4 validator nodes.  Org B has 3 Tx , 3 Tessera & 1 validator node. 

**3 Tessera Node = ( 1 Actual Tessera node + 2 Replicas of it) for each organization and 3 Transaction nodes of each Org are using same Tessera node.** 

When private tx is performed by either party i.e Org A, its tx data including events emitted are replicated on all tx nodes of Org A. However, it is not getting replicated on all the tx nodes of Org B, its only visible on one of the node. As a result, unable to get the data from Org B's other tx nodes.  The vice-versa is also true.

**Note**:  Check the below attached screenshot for the same. Both the screenshot are of same Org nodes which shows transaction receipt for a tx  that has event data on Node1 but  its not there on node 2.

**Node 1:** 
![OrgA-node1-1919339](https://user-images.githubusercontent.com/18182040/185984978-89604f7d-83e4-41ad-b335-725214492298.png)

**Node 2:**
![OrgB-node2-1919339](https://user-images.githubusercontent.com/18182040/185984988-ddee0be6-3c96-45f0-94ed-b6fbb2875dee.png)

**Comment by baptiste-b-pegasys** [comment]
hello, do you use the same version and flags on every node?
what represent node1 and node2 in the org? (who is validator?)

**Comment by rushikeshacharya** [comment]
Hi, I do have same version and flags on every node. Both Node 1 and 2 are transaction nodes of same Org. One node has the event data and second does not have it. So whenever the get request goes to the first node, we don't get the data.

**Comment by antonydenyer** [comment]
Can you provide details of how you are sending the transaction? ie what have you got in privateFor?

**Reply by rushikeshacharya** [reply]
We are passing tessera public keys of all participants involved in the tx. Below is the sample transactionObject.
`transactionObject = {
        from: quorumAccountAddress,
        gasPrice: BC_DEFAULTS.GAS_PRICE,
        gas: BC_DEFAULTS.GAS,
        privateFor: [nodeATesseraPK, nodeBTesseraPK],
        privacyFlag: 3
    };`

We are explicitly signing the tx and  invoking sendRawPrivateTransaction method of web3js-quorum NodeJS client.

**Reply by antonydenyer** [reply]
Just to confirm the only nodes that will see this are `nodeA` that is connected to tessera that has `nodeATesseraPK` and `nodeB` that is connected to `nodeBTesseraPK`? And that `nodeB` is not seeing the transaction?

**Reply by rushikeshacharya** [reply]
Actually, we have **only one Tessera node** for each organization and other **2 tessera nodes are the replicas of it**. So, all three transaction nodes are using same Tessra node..

`transactionObject = { from: quorumAccountAddress, gasPrice: BC_DEFAULTS.GAS_PRICE, gas: BC_DEFAULTS.GAS, privateFor: [OrgATesseraPK, OrgBTesseraPK], privacyFlag: 3 };`
